### PR TITLE
Automatically register the metics of `PooledByteBufAllocator.DEFAULT`

### LINKED ISSUE #2633: Add metrics about `PooledByteBufAllocator`
Netty already exposes the metrics about its `PooledByteBufAllocator`. We could expose that information into Armeria's Micrometer metrics: https://github.com/netty/netty/blob/4.1/buffer/src/main/java/io/netty/buffer/PooledByteBufAllocatorMetric.java

### PR DESCRIPTION
**Pull request by Bue-von-hon**
Motivation:
Add metrics related to PooledByteBufAllocator that is already exposed by the netty.

Modifications:
- `NettyAllocatorMetrics` are registered when MoreMeterBinders is initialized.

Result:

- Fixes #2633
- Armeria now automatically registers the metics of `PooledByteBufAllocator.DEFAULT`.

### CODE REVIEW

**Comment by jrhee17** [comment]
Micrometer already provides bindings to Netty Metrics (ref: https://docs.micrometer.io/micrometer/reference/reference/netty.html)

I prefer that we either:
1) We close this issue and allow users to add metric bindings themselves. Adding the metric bindings is not difficult and the scope of applying this metric seems ambiguous
2) We add bindings for servers only by default. The name will probably be a combination of local address + ports.

Let me know what you think @line/dx

**Comment by ikhoon** [comment]
I didn't know that Netty metrics were supported by Micrometer finally. 

I prefer automatically setting the metric for the default one (`ByteBufAllocator.DEFAULT`) when the `MoreMeterBinders` class is loaded.

**Comment by Bue-von-hon** [comment]
@ikhoon @jrhee17 @trustin @minwoox 
I trust this implementation satisfies your requirements.
If the goal was to consolidate PooledByteBufAllocatorMetrics with eventLoopMetrics or certificateMetrics that would involve substantial modifications. (However, I don't believe that was the intended direction 😅)

**Comment by Bue-von-hon** [comment]
@ikhoon @jrhee17 @trustin @minwoox gentle ping.

**Comment by ikhoon** [comment]
I think my intention was not conveyed. Unlike when you first created this PR, Micrometer now supports Netty metrics. Let me add a commit to apply that.

**Comment by ikhoon** [comment]
Changed the title of this PR to reflect the purpose. 🙇‍♂️

**Review by ikhoon — APPROVED** [review]
Thanks, @Bue-von-hon 👍🙏

**Comment by Bue-von-hon** [comment]
@jrhee17 @trustin @minwoox gentle ping.

**Comment by Bue-von-hon** [comment]
@jrhee17 @trustin @minwoox gentle ping.

File: core/src/main/java/com/linecorp/armeria/common/metric/MoreMeterBinders.java [hunk 1]
```diff
@@ -25,17 +25,26 @@
...
 import io.netty.channel.EventLoopGroup;
 
 /**
  * Provides useful {@link MeterBinder}s to monitor various Armeria components.
  */
 public final class MoreMeterBinders {
 
+    static {
+        // Bind the default Netty allocator metrics to the default MeterRegistry.
+        new NettyAllocatorMetrics(PooledByteBufAllocator.DEFAULT)
```
Comment (jrhee17) [inline core/src/main/java/com/linecorp/armeria/common/metric/MoreMeterBinders.java:44]: Note: `armeria` always initializes this class when a client or server is built

**Review by jrhee17 — APPROVED** [review]

**Review by minwoox — APPROVED** [review]
Thanks!

**Comment by ikhoon** [comment]
@Bue-von-hon Thanks for your effort. That said maintainers are working on various issues and are aware of the status of this PR, so you may not need to keep mentioning us. 🙏

**Pull request merged**